odoo/addons/base/models/ir_model.py

```diff
@@ -1012,10 +1012,10 @@ def _reflect_fields(self, model_names):
             by_label = {}
             for field in model._fields.values():
                 if field.string in by_label:
-                    _logger.warning('Two fields (%s, %s) of %s have the same label: %s.',
-                                    field.name, by_label[field.string], model, field.string)
+                    _logger.warning('Two fields (%s, %s) of %s have the same label: %s. [Modules: %s and %s]',
+                                    field.name, by_label[field.string][0], model, field.string, field._module, by_label[field.string][1])
                 else:
-                    by_label[field.string] = field.name
+                    by_label[field.string] = [field.name, field._module]
```

Comment:
Wouldn't it be simpler to put the field itself in `by_label`?
```suggestion
                    by_label[field.string] = field
```